### PR TITLE
Optimize prune unused fields

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
    NONE
```

### CODE REVIEW

File: pkg/resourceinterpreter/defaultinterpreter/prune/prune.go [hunk 1]
```diff
@@ -14,38 +14,10 @@ import (
...
-	unstructured.RemoveNestedField(workload.Object, "metadata", "generation")
-
-	// This is mostly for internal housekeeping, and users typically shouldn't need to set or understand this field.
-	// Remove this field to keep 'Work' clean and tidy.
-	unstructured.RemoveNestedField(workload.Object, "metadata", "managedFields")
-
-	// populated by the kubernetes.
-	unstructured.RemoveNestedField(workload.Object, "metadata", "resourceVersion")
-
-	// populated by the kubernetes and has been deprecated by kubernetes.
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/prune/prune.go:40]: To be honest, I can't see much benefit from this change. In addition, I hope this kind of comment is still useful and should not be removed.

Comment (lonelyCZ) [reply pkg/resourceinterpreter/defaultinterpreter/prune/prune.go:40]: I think so, these comments is useful for me to understand why to remove these fields.

Comment (pigletfly) [reply pkg/resourceinterpreter/defaultinterpreter/prune/prune.go:40]: Comments can left above getIrrelevantFields func.The benefit is the removal of redundant code,if we want to add more fields to ignore,we have to copy `unstructured.RemoveNestedField(workload.Object, "metadata", "uid")` and paste.

**Comment by RainbowMango** [comment]
sorry, we need more evidence for the benefits.
/close

**Pull request closed without merging**